src/hayhooks/server/utils/deploy_utils.py

```diff
@@ -370,6 +301,58 @@ def add_pipeline_api_route(app: FastAPI, pipeline_name: str, pipeline_wrapper: B
     app.setup()
 
 
+def add_pipeline_yaml_api_route(app: FastAPI, pipeline_name: str) -> None:
+    """
+    Create or replace the YAML pipeline run endpoint at /{pipeline_name}/run.
+
+    Builds the flat request/response models from declared YAML inputs/outputs and wires a handler that
+    maps the flat body into the nested structure required by Haystack Pipeline.run.
+    """
+    pipeline_instance = registry.get(pipeline_name)
+    if pipeline_instance is None:
+        msg = f"Pipeline '{pipeline_name}' not found"
+        raise PipelineNotFoundError(msg)
+
+    # Ensure the registered object is a Haystack Pipeline, not a wrapper
+    if not isinstance(pipeline_instance, AsyncPipeline):
+        msg = f"Pipeline '{pipeline_name}' is not a Haystack AsyncPipeline instance"
+        raise PipelineYamlError(msg)
```

Comment:
Dev comment doesn't quite line up with the actual check which is that it must be an AsyncPipeline and not a normal Pipeline?

Reply:
Does this also mean this deploying with yaml only works with `AsyncPipeline`?